mmcls/engine/hooks/tau_norm_hook.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+import torch
+from mmengine.hooks import Hook
+
+from mmcls.registry import HOOKS
+
+
+def pnorm(weights, p):
+    normB = torch.norm(weights, 2, 1)
+    ws = weights.clone()
+    for i in range(weights.size(0)):
+        ws[i] = ws[i] / torch.pow(normB[i], p)
+    return ws
+
+
+@HOOKS.register_module()
+class TauNormHook(Hook):
```

Comment:
Can this hook be moved into the `LWSHead.predict`? The use of an additional hook is not user-friendly

Reply:
During inference, we only need apply Tau Norm for head weights one time and then use these normed weights for all test images. If move this hook to LWSHead.predict, for each image we should calculate the norm weights, it's not efficient.

Reply:
I think we can use the train, eval function in `nn.Module`. The definition is that there are two parameters, train_p and eval_p.

```
def eval():
    eval_p = pnorm(train_p)
    ........


def forward():
    if mode.training:
         # use train_p
    else:
         # use eval_p
```

how about this design?

Reply:
It's a good idea and I have implemented it.